### PR TITLE
Switch from dep to go modules

### PR DESCRIPTION
**Pull request by steakunderscore**
## Description

Switch from [go dep](https://github.com/golang/dep) to [go modules](https://github.com/golang/go/wiki/Modules).

I also ended up upgrading most modules to avoid issues with golangci-lint.

## Motivation and Context

I figured it would be wise to make this transissions before go 1.13 is release and enabling go modules is the default.

## How Has This Been Tested?

Smoke tested on GKE

## Checklist:

- [X] My change requires a change to the documentation or CHANGELOG.
- [X] I have updated the documentation/CHANGELOG accordingly.
- [X] I have created a feature (non-master) branch for my PR.

### CODE REVIEW

**Comment by dekimsey** [comment]
> Is there an equivalent caching we can achieve as described in #209?

I don't see why not, should be as straightforward as:

```
COPY go.mod go.sum ./
RUN go mod download
COPY . .
RUN go build ...
```

**Comment by steakunderscore** [comment]
What @dekimsey has should do the trick. @dekimsey want to rebase off of my branch? Or would you prefer I merge your change in to my changes? Either way I'll fix this up and test it later today.

**Comment by JoelSpeed** [comment]
@steakunderscore I'll merge #209 now, then can you rebase on top of that? I think that then gives @dekimsey his accreditation for it and shows the transition between the two in a nicer way. The caching improvements are separated from the migration

File: Makefile [hunk 1]
```diff
@@ -17,17 +17,13 @@ distclean: clean
 
 .PHONY: lint
 lint:
-	$(GOLANGCILINT) run
-
-.PHONY: dep
-dep:
-	$(DEP) ensure --vendor-only
+	GO111MODULE=on $(GOLANGCILINT) run
```
Comment (JoelSpeed) [inline Makefile]: If there is no vendor folder present, what happens when you run `make lint`? Will it download the modules?

Comment (steakunderscore) [reply Makefile]: Had to verify that one. It works as expected, downloading the modules for before linting.

I tested by `rm -rf $GOPATH/pkg/*` then `./configure && make lint`.

**Pull request merged**